include/libsemigroups/bipart.hpp

```diff
@@ -383,31 +705,60 @@ namespace libsemigroups {
       LIBSEMIGROUPS_ASSERT(i < _blocks.size());
       return _blocks[i];
     }
-  };
 
-  //! Validates a Blocks object.
-  //!
-  //! \param x the blocks object to validate.
-  //!
-  //! \returns
-  //! (None)
-  //!
-  //! \throws LibsemigroupsException if \p x is invalid.
-  void validate(Blocks const& x);
+    //! \brief Return a const reference to the index of the block containing a
+    //! point.
+    //!
+    //! \param i the point.
+    //!
+    //! \returns A value const reference to a value of type \c uint32_t.
+    //!
+    //! \throws std::out_of_range if \p i is out of range.
+    //!
+    //! \complexity
+    //! Constant.
+    [[nodiscard]] uint32_t const& at(size_t i) const {
+      // TODO(2) better error
+      return _blocks.at(i);
+    }
 
-  // Forward decl
-  class Bipartition;
+    //! \brief Validate the arguments, construct a Blocks object, and validate
+    //! it.
+    //!
+    //! \tparam Container the type of the parameter \p cont
+    //!
+    //! \param cont container containing a lookup for the blocks.
+    //!
+    //! \throws LibsemigroupsException if the arguments do not describe a
+    //! signed partition.
+    //!
+    //! \throws LibsemigroupsException if the constructed Blocks object is not
+    //! valid.
+    template <typename Container>
+    [[nodiscard]] static Blocks make(Container const& cont);
 
-  //! Validates a bipartition.
-  //!
-  //! \param x the bipartition
+    //! \copydoc make(Container const&)
+    [[nodiscard]] static Blocks
+    make(std::initializer_list<std::vector<int32_t>> const& cont) {
+      return make<std::initializer_list<std::vector<int32_t>>>(cont);
+    }
```

Comment:
Should this be called `to_blocks` like with presentations?

Reply:
Yeah you're almost certainly correct, I thought having `Blocks::make` was close enough but I see I am just being lazy :)